### PR TITLE
Allowing reserved keywords, since PHP 7.0

### PR DESCRIPTION
**Pull request by MartinPham**
Starting from PHP 7.0, some keywords can be used. So we exclude from list of keywords, and only add when PHP version < 7.0

### CODE REVIEW

**Comment by stof** [comment]
I think your list of keywords being allowed on PHP 7 is incomplete here. Many more keywords are now allowed in method names AFAIK.

I'm in favor of accepting such changes, but only if done in a complete way, not only for ``list``

**Comment by MartinPham** [comment]
@stof I've updated the code, thanks!

File: src/Prophecy/Doubler/ClassPatch/KeywordPatch.php [hunk 1]
```diff
@@ -61,75 +61,87 @@ public function getPriority() {
...
-            'trait',
-            'try',
-            'unset',
-            'use',
-            'var',
-            'while',
-            'xor',
-            'yield',
+        $keywords = array(
+            '__halt_compiler'
```
Comment (stof) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: is this really the only forbidden keyword on PHP 7 ? If yes, I would rather do this:

```php
if (\PHP_VERSION_ID >= 70000) {
    return array('__halt_compiler');
}

return array(
    '__halt_compiler',
    // ...
);
```

It would keep the code much more readable.

Comment (MartinPham) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: Yea, I thought about it also (it is bad to append keywords many times), but I'm worry next versions will have some more reserved keywords.

**Comment by MartinPham** [comment]
@stof I've modified it based on your idea, thanks!

File: src/Prophecy/Doubler/ClassPatch/KeywordPatch.php [hunk 2]
```diff
@@ -130,6 +139,6 @@ private function getKeywords() {
             'while',
             'xor',
             'yield',
-        );
+	    );
```
Comment (stof) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: can you revert this change ?

File: src/Prophecy/Doubler/ClassPatch/KeywordPatch.php [hunk 3]
```diff
@@ -61,6 +61,15 @@ public function getPriority() {
      * @return array
      */
     private function getKeywords() {
+        /*
+         * Starting from PHP 7.0, almost keywords can be used
```
Comment (avant1) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: Probably typo: "almost **all** keywords"

**Comment by MartinPham** [comment]
@stof @avant1 done :)

File: src/Prophecy/Doubler/ClassPatch/KeywordPatch.php [hunk 4]
```diff
@@ -129,7 +138,7 @@ private function getKeywords() {
             'var',
             'while',
             'xor',
-            'yield',
-        );
+            'yield'
+	);
```
Comment (avant1) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: I think stof wanted this line to be not changed in this PR.
Now there is tab character instead of 4 spaces.

**Comment by avant1** [comment]
@MartinPham can you rebase this changes on master? Build should be fixed in master now.

**Comment by everzet** [comment]
Please rebase and try to fix the build.

File: src/Prophecy/Doubler/ClassPatch/KeywordPatch.php [hunk 5]
```diff
@@ -129,7 +138,7 @@ private function getKeywords() {
             'var',
             'while',
             'xor',
-            'yield',
-        );
+            'yield'
```
Comment (stof) [inline src/Prophecy/Doubler/ClassPatch/KeywordPatch.php]: the trailing comma should not be removed either

**Comment by nesl247** [comment]
Any chance we can get this merged in? Just ran into this issue and it's a big blocker for me.

**Comment by olvlvl** [comment]
WANT! I just created this issue: https://github.com/phpspec/prophecy/issues/351

**Comment by Taluu** [comment]
was already taken care of in https://github.com/phpspec/prophecy/pull/337
